### PR TITLE
clippy: Fix `type_complexity` warning in `components/script/dom`

### PR DESCRIPTION
**Pull request by taniishkaaa**
Fixes `type_complexity` warning in `components/script/dom/bluetoothdevice.rs` as a part of #31500 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are a part of #31500 
- [X] These changes do not require tests because they do not modify functionality

### CODE REVIEW

**Review by jdm — APPROVED** [review]
Looks like a reasonable cleanup! Thanks!

File: components/script/dom/bluetoothdevice.rs [hunk 1]
```diff
@@ -34,18 +34,21 @@ use crate::dom::promise::Promise;
 use crate::realms::InRealm;
 
 // https://webbluetoothcg.github.io/web-bluetooth/#bluetoothdevice
```
Comment (jdm) [inline components/script/dom/bluetoothdevice.rs:36]:
```suggestion
```

File: components/script/dom/bluetoothdevice.rs [hunk 2]
```diff
@@ -34,18 +34,21 @@ use crate::dom::promise::Promise;
...
+#[derive(JSTraceable, MallocSizeOf)]
+struct AttributeInstanceMap {
+    service_map: DomRefCell<HashMap<String, Dom<BluetoothRemoteGATTService>>>,
+    characteristic_map: DomRefCell<HashMap<String, Dom<BluetoothRemoteGATTCharacteristic>>>,
+    descriptor_map: DomRefCell<HashMap<String, Dom<BluetoothRemoteGATTDescriptor>>>,
+}
+
 #[dom_struct]
```
Comment (jdm) [inline components/script/dom/bluetoothdevice.rs:44]:
```suggestion
// https://webbluetoothcg.github.io/web-bluetooth/#bluetoothdevice
#[dom_struct]
```

**Review by jdm — CHANGES_REQUESTED** [review]
Sorry, just noticed that the changes made a documentation comment move further away from the code is was related to.

**Comment by taniishkaaa** [comment]
> Sorry, just noticed that the changes made a documentation comment move further away from the code is was related to.

updated 👍

**Review by jdm — APPROVED** [review]

**Review by mukilan — APPROVED** [review]
Thanks!

**Pull request merged**